inject/src/main/java/io/micronaut/context/event/BeanCreatedEvent.java

```diff
@@ -31,15 +32,23 @@ public class BeanCreatedEvent<T> extends BeanEvent<T> {
 
     private final BeanIdentifier beanIdentifier;
 
+    private final Argument<T> argument;
+
     /**
      * @param beanContext    The bean context
      * @param beanDefinition The bean definition
      * @param beanIdentifier The bean identifier
+     * @param argument       The {@link Argument} used to create the bean
      * @param bean           The bean
      */
-    public BeanCreatedEvent(BeanContext beanContext, BeanDefinition<T> beanDefinition, BeanIdentifier beanIdentifier, T bean) {
+    public BeanCreatedEvent(BeanContext beanContext,
+                            BeanDefinition<T> beanDefinition,
+                            BeanIdentifier beanIdentifier,
+                            Argument<T> argument,
+                            T bean) {
```

Comment:
this is a breaking API change. You will need to add a new constructor with the argument (I would call the parameter beanType) and delegate to this new constructor. You can use `beanDefinition.asArgument()` as a default value to maintain compatibility.

Reply:
Thanks for catching that. Did not consider that user code could also create instances of this